caddyhttp/httpserver/server.go

```diff
@@ -507,16 +507,34 @@ func (s *Server) Stop() error {
 // OnStartupComplete lists the sites served by this server
 // and any relevant information, assuming caddy.Quiet == false.
 func (s *Server) OnStartupComplete() {
-	if caddy.Quiet {
-		return
+	if !caddy.Quiet {
+		firstSite := s.sites[0]
+		scheme := "HTTP"
+		if firstSite.TLS.Enabled {
+			scheme = "HTTPS"
+		}
+		fmt.Println("")
+		fmt.Printf("Serving %s on port "+firstSite.Port()+" \n", scheme)
+		s.OutputSiteInfo("fmt")
+	}
+	// if caddy process log is going to stdout, printing to log
+	// here would duplicate to stdout so dont.
+	if caddy.LogDestination != "stdout" {
```

Comment:
If we have `caddy.Quiet = true` and `caddy.LogDestination = "stdout"` here, we will end up printing nothing.

This check should probably be `if caddy.LogDestination != "stdout" || caddy.Quiet` so we always print at least one set of addresses (but still avoid doubling up).

Reply:
I understand what you are saying, but surely that's whats wanted.

> caddy -log stdout -quiet

If you call this I would expect no output.  It seems a bit odd to print to the screen when -quiet says it wont

> Quiet mode (no initialization output)

Reply:
hmm but yes I see you get other output.  I still think if you call

> caddy -log stdout -quiet 

You've done something wrong 😈

Reply:
I think `-quiet` should be irrespective of where you designate your process log to be sent.

The check should only be there to ensure that we don't _double up_ on printouts; if you don't want your _log_ output to `stdout`, you'd have sent it elsewhere; the log output should never be suppressed.

Reply:
There's also the fact that having `-quiet` not suppress the log gives the user the ability to ensure either nothing gets printed to stdout, or the log output gets printed to stdout, or the startup grouped output gets printed out. The user can, with the correct options, get any desired output. With your method, there's no way to ensure that `stdout` is the only output and that the regular output isn't included.

Reply:
I think I agree with @whitestrake. We could just add a note on the `-quiet` docs to clarify that edgecase. I think `-log` wins over `-quiet`.

Reply:
Yeah, specifically, I don't consider `log` output to be "initialization" output.

Reply:
OK - I'll make the update